app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java

```diff
@@ -87,15 +99,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             mapView = binding.mapView;
             mapView.onCreate(savedInstanceState);
             mapView.getMapAsync(this);
-
             initializeAvailableColors();
 
             binding.walkingRoutesFilterScrollView.setAlpha(TRANSPARENCY);
         }
 
+        flSharePetWalkRouteButton = binding.flSharePetWalkRouteButton;
+        setUpSharePetWalkRouteListener();
         return binding.getRoot();
     }
 
+    /**
+     * Set the share app button listener.
+     */
+    private void setUpSharePetWalkRouteListener() {
+        flSharePetWalkRouteButton.setOnClickListener(v -> {
+            GoogleMap.SnapshotReadyCallback callback = new GoogleMap.SnapshotReadyCallback() {
+            @Override
+                public void onSnapshotReady(Bitmap bitmap) {
+                    saveImage(bitmap);
+                }
+            };
+            googleMap.snapshot(callback);
+        });
+    }
+
+
+    /**
+     * Saves the image as PNG to the app's cache directory and share.
+     * @param image Bitmap to share.
+     */
+    private void saveImage(Bitmap image) {
+        File file = saveBitmap(image);
+        Uri uri = Uri.fromFile(file);
+        shareUri(uri);
+    }
+
+    /**
+     * Creates a intent to share the uri.
+     * @param uri The uri to share
+     */
+    private void shareUri(Uri uri) {
+        Intent intent = new Intent();
+        intent.setAction(Intent.ACTION_SEND);
+        intent.setType("image/jpeg");
+
+        intent.putExtra(Intent.EXTRA_SUBJECT, "My Pet Care");
+        intent.putExtra(Intent.EXTRA_TEXT, "");
+        intent.putExtra(Intent.EXTRA_STREAM, uri);
+        startActivity(intent);
+    }
+
+    /**
+     * Save a bitmap.
+     * @param image The bitmap of the image
+     * @return The file created
+     */
+    @NonNull
+    private File saveBitmap(Bitmap image) {
+        File file = fileCreation();
+        FileOutputStream fOut;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this FileOutputStream object are closed after use](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)